ui/packages/tidb-dashboard-lib/src/apps/Statement/pages/Detail/PlanBind.tsx

```diff
@@ -0,0 +1,292 @@
+    if (plan.length === 0) return setSelectedPlan(null)
+    return setSelectedPlan(plan[0].plan_digest!)
+  }
+
+  return (
+    <Modal
+      visible={showPlanBindModal}
+      title={
+        <Space direction="vertical">
+          <Space size={10}>
+            {t('statement.pages.detail.plan_bind.title')}{' '}
+            {boundPlanDigest ? (
+              <Space className={`${styles.SmallFont}`}>
+                <span className={styles.GreenDot} />
+                {t('statement.pages.detail.plan_bind.bound')}
+              </Space>
+            ) : (
+              <Space className={`${styles.SmallFont}`}>
+                <span className={styles.GreyDot} />
+                {t('statement.pages.detail.plan_bind.not_bound')}
+              </Space>
+            )}
+          </Space>
+          <Alert
+            type="warning"
+            message={t('statement.pages.detail.plan_bind.notice')}
+            className={`${styles.SmallFont}`}
+          />
+        </Space>
+      }
+      onCancel={() => onHandleModalVisibility(false)}
+      width={1000}
+      footer={
+        <div className={styles.Center}>
+          {boundPlanDigest ? (
+            <Space direction="vertical">
+              {t('statement.pages.detail.plan_bind.bound_status_desc')}
+              <Button
+                onClick={handleDropPlan}
+                loading={isLoading}
+                disabled={isLoading}
+              >
+                {t('statement.pages.detail.plan_bind.drop_btn_txt')}
+              </Button>
+            </Space>
+          ) : (
+            <Button
+              onClick={handlePlanBind}
+              loading={isLoading}
+              disabled={isLoading || !selectedPlan}
+            >
+              {t('statement.pages.detail.plan_bind.bind_btn_txt')}
+            </Button>
+          )}
+        </div>
+      }
+      destroyOnClose
+    >
+      <p>{t('statement.pages.detail.plan_bind.bound_sql')}</p>
+      <pre className={`${styles.PreBlock} ${styles.SmallFont}`}>
+        {sqlDigest}
+      </pre>
+      <p>{t('statement.pages.detail.plan_bind.to_plan')}</p>
+      {!isLoading && (
+        <PlanTable
+          plans={plans}
+          boundPlanDigest={boundPlanDigest}
+          onHandleSelectedPlanChange={handleSelectedPlanChange}
+        />
+      )}
+    </Modal>
+  )
+}
+
+interface PlanTableProps {
+  boundPlanDigest: string | null
+  plans: StatementModel[]
+  onHandleSelectedPlanChange: (plan: StatementModel[] | []) => void
+}
+
+const PlanTable = ({
+  boundPlanDigest,
+  plans,
+  onHandleSelectedPlanChange
+}: PlanTableProps) => {
+  const planColumns = useMemo(() => genPlanColumns(plans || []), [plans])
+
+  const selection = useRef(
+    new Selection({
+      canSelectItem: (item) => {
+        const digest = (item as StatementModel).plan_digest
+        return !boundPlanDigest
+          ? true
+          : digest === boundPlanDigest
+          ? true
+          : false
```

Comment:
Can be simplified to:

`return !boundPlanDigest || digest === boundPlanDigest`